database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -47,76 +41,95 @@ public void cleanup() {
 
     public int getCount() {
         return mSnapshots.size();
-
     }
+
     public DataSnapshot getItem(int index) {
         return mSnapshots.get(index);
     }
 
+    protected DataSnapshot internalGetItem(int index) {
```

Comment:
This is completely gross IMO, but I don't know how to get rid of it. Any feedback is welcome!